satpy/cf/coords.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Set CF-compliant spatial and temporal coordinates."""
+
+import logging
+import warnings
+from collections import defaultdict
+from contextlib import suppress
+
+import numpy as np
+import xarray as xr
+from dask.base import tokenize
+from pyresample.geometry import AreaDefinition, SwathDefinition
+
+logger = logging.getLogger(__name__)
+
+
+def add_xy_coords_attrs(dataarray):
+    """Add relevant attributes to x, y coordinates."""
+    # If there are no coords, return dataarray
+    if not dataarray.coords.keys() & {"x", "y", "crs"}:
+        return dataarray
+    # If projected area
+    if _is_projected(dataarray):
+        dataarray = _add_xy_projected_coords_attrs(dataarray)
+    else:
+        dataarray = _add_xy_geographic_coords_attrs(dataarray)
+    if "crs" in dataarray.coords:
+        dataarray = dataarray.drop_vars("crs")
+    return dataarray
+
+
+def _is_projected(dataarray):
+    """Guess whether data are projected or not."""
+    crs = _try_to_get_crs(dataarray)
+    if crs:
+        return crs.is_projected
+    units = _try_get_units_from_coords(dataarray)
+    if units:
+        if units.endswith("m"):
+            return True
+        if units.startswith("degrees"):
+            return False
```

Comment:
These if statements are driving me nuts, but I can't think how they can be combined with the warning below. I really want `return units.endswith("m")` or something, but...I don't know. Just wanted to get this out of my head.

Reply:
Can this be done with pattern matching? https://peps.python.org/pep-0634/
(Real question — I don't know the answer)

Reply:
Hm probably. I've only used match/case statements in rust and only seen them in presentations for python. I'm not sure if you have to do anything special to handle the endswith/startswith method calls.

Reply:
Ah but we still support Python 3.9 (right?) and this is 3.10+.

Reply:
Do we still support Python 3.9?  If we follow https://scientific-python.org/specs/spec-0000/ then the recommendation would be to no longer support it.

Reply:
I mean, we technically do:

https://github.com/pytroll/satpy/blob/6fb1d1ea4f3c6e95cd8178d9989519f557a51366/setup.py#L166

I'm OK dropping it, but I also don't remember if Python 3.12 is fully available for our Satpy environments on conda-forge yet.

Reply:
I didn't address this for the moment !!!
In case let me know what to do ...